### PR TITLE
chore: eslint fixes for content scripts

### PR DESCRIPTION
**Pull request by 0xmad**
## Explanation

This is a third PR for new linter config. There are some fixes for content scripts.

## More Information

Related to #142 
Blocked by #170 

## Screenshots/Screencaps

N/A

## Manual Testing Steps

N/A

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

> PR template source from [github.com/MetaMask](https://github.com/MetaMask)

### CODE REVIEW

File: src/contentScripts/index.ts [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+  scriptTag.src = url;
+  scriptTag.setAttribute("async", "false");
+  container.insertBefore(scriptTag, container.children[0]);
+  container.removeChild(scriptTag);
 
-    window.addEventListener("message", async (event) => {
-      const { data } = event;
-      if (data && data.target === "injected-contentscript") {
-        const res = await browser.runtime.sendMessage(data.message);
+  window.addEventListener("message", (event: MessageEvent<{ target: string; message: unknown; nonce: string }>) => {
```
Comment (iskdrews) [inline src/contentScripts/index.ts:17]: Could we move this type better in the `@src/types/index.ts`?

File: src/contentScripts/injected.ts [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-import { MerkleProofArtifacts } from "@src/types";
 import { RPCAction } from "@src/constants";
+import { MerkleProofArtifacts } from "@src/types";
+
 import { IRlnGenerateArgs, ISemaphoreGenerateArgs, RlnProofGenerator, SemaphoreProofGenerator } from "./proof";
 
 export type IRequest = {
   method: string;
-  payload?: any;
+  payload?: unknown;
```
Comment (iskdrews) [inline src/contentScripts/injected.ts:11]: Are we sure that the payload is `unknown`?

File: src/contentScripts/injected.ts [hunk 3]
```diff
@@ -185,12 +184,10 @@ const off = (eventName: string, cb: (data: unknown) => void) => {
   EVENTS[eventName] = bucket.filter((callback) => callback === cb);
 };
 
-const emit = (eventName: string, payload?: any) => {
+const emit = (eventName: string, payload?: unknown) => {
```
Comment (iskdrews) [inline src/contentScripts/injected.ts:187]: Are we sure that the payload is `unknown`?

**Review by iskdrews — CHANGES_REQUESTED** [review]
Thanks @0xmad just some comments then we are good to merge.

File: src/contentScripts/injected.ts (hunk 2 above)
Comment (0xmad) [reply src/contentScripts/injected.ts:11]: Yes, this is a general method which can take any payload type.

File: src/contentScripts/index.ts (hunk 1 above)
Comment (0xmad) [reply src/contentScripts/index.ts:17]: I'll fix

File: src/contentScripts/injected.ts (hunk 3 above)
Comment (0xmad) [reply src/contentScripts/injected.ts:187]: I think it's the same because we can emit any payload.

**Review by iskdrews — APPROVED** [review]
Thanks @0xmad

**Pull request merged**